Release 1.6.0 (#3531)
antrea-io/antrea: CHANGELOG/CHANGELOG-1.6.md

# Changelog 1.6

## 1.6.0 - 2022-03-29

The Egress feature is graduated from Alpha to Beta and is therefore enabled by default.

### Added

- Add support for the following capabilities for [Antrea IPAM] feature:
  * Support pre-allocating continuous IPs for StatefulSet. ([#3281](https://github.com/antrea-io/antrea/pull/3281), [@annakhm])

>   * Support specifying VLAN for IPPool. Traffic from the Pods whose IPPool is configured with VLAN ID will be tagged when leaving Node uplink. ([#3247](https://github.com/antrea-io/antrea/pull/3247), [@gran-vmv])

Reviewer: "from a Pod whose IPPool is" or "from Pods whose IPPools are"

"with a VLAN ID"

"the Node uplink"

- Add support for the following capabilities for [Antrea Multi-cluster] feature:

>   * Add ACNP copy span for multi-cluster. ([#3363](https://github.com/antrea-io/antrea/pull/3363), [@Dyanngg])

Reviewer: Do not call it "copy span" as it is not a well-known term. Say: "Support Antrea ClusterNetworkPolicy replication in a ClusterSet"

>   * Add `antctl mc` subcommand for antrea multicluster resources. ([#3341](https://github.com/antrea-io/antrea/pull/3341) [#3287](https://github.com/antrea-io/antrea/pull/3287), [@hjiajing] [@bangqipropel])

Reviewer: Antrea Multi-cluster

- Add support for the following capabilities for [AntreaPolicy] feature:

>   * Add Node selector in Antrea-native policies to allow matching traffic originating from or intended for Nodes. ([#3038](https://github.com/antrea-io/antrea/pull/3038), [@wenqiq])

Reviewer: "intended for" -> "destined for"

  * Add ServiceAccount selector in Antrea-native policies to allow selecting Pods by ServiceAccount. ([#3044](https://github.com/antrea-io/antrea/pull/3044), [@GraysonWu])
  * Support Pagination for ClusterGroupMembership. ([#3183](https://github.com/antrea-io/antrea/pull/3183), [@qiyueyao])
  * Add Port Number to Audit Logging. ([#3277](https://github.com/antrea-io/antrea/pull/3277), [@qiyueyao])

> - [Flow Visibility] Add support for exporting flow records to ClickHouse. ([#3196](https://github.com/antrea-io/antrea/pull/3196) [#3498](https://github.com/antrea-io/antrea/pull/3498) [#3244](https://github.com/antrea-io/antrea/pull/3244) [#3063](https://github.com/antrea-io/antrea/pull/3063) [#3312](https://github.com/antrea-io/antrea/pull/3312), [@wsquan171] [@yanjunz97] [@zyiou] [@dreamtalen])

Reviewer 2: Thanks Quan, I'm going to add some details here:
`
- [Flow Visibility] Add Grafana Flow Collector as the new visualization tool for flow records.
  * Add Grafana dashboards, Clickhouse data schema, deployment files, and doc. [#3063][#3525] @heanlan @zyiou @dreamtalen 
  * Add support for exporting flow records to ClickHouse from Flow Aggregator. [#3196][#3526] @wsquan171 @dreamtalen 
  * Add ClickHouse monitor to ensure data retention for in-memory ClickHouse deployment [#3244][#3498] @yanjunz97 
  * Add CI validation job for Flow Visibility deployment [#3312]@dreamtalen 

`
cc. @heanlan to review this too.

> - [Flow Visibility] Add support for exporting flow records to ClickHouse. ([#3196](https://github.com/antrea-io/antrea/pull/3196) [#3498](https://github.com/antrea-io/antrea/pull/3498) [#3244](https://github.com/antrea-io/antrea/pull/3244) [#3063](https://github.com/antrea-io/antrea/pull/3063) [#3312](https://github.com/antrea-io/antrea/pull/3312), [@wsquan171] [@yanjunz97] [@zyiou] [@dreamtalen])

Author: Done, but removed the 4th item as it's not user facing.

> - [Flow Visibility] Add support for exporting flow records to ClickHouse. ([#3196](https://github.com/antrea-io/antrea/pull/3196) [#3498](https://github.com/antrea-io/antrea/pull/3498) [#3244](https://github.com/antrea-io/antrea/pull/3244) [#3063](https://github.com/antrea-io/antrea/pull/3063) [#3312](https://github.com/antrea-io/antrea/pull/3312), [@wsquan171] [@yanjunz97] [@zyiou] [@dreamtalen])

Reviewer 3: Looks good to me. Thanks @dreamtalen for the feedback.

  * Add [documentation](https://github.com/antrea-io/antrea/blob/v1.6.0/docs/antrea-ipam.md) for Multiple-VLAN support. ([#3507](https://github.com/antrea-io/antrea/pull/3507), [@gran-vmv])
  * Add [upgrade guide](https://github.com/antrea-io/antrea/blob/v1.6.0/docs/multicluster/upgrade.md) for Multi-cluster. ([#3374](https://github.com/antrea-io/antrea/pull/3374), [@luolanzone])
  * Add [a per-rule example](https://github.com/antrea-io/antrea/blob/v1.6.0/docs/feature-gates.md#networkpolicystats) for NetworkpolicyStats docs . ([#3356](https://github.com/antrea-io/antrea/pull/3356), [@ceclinux])

### Changed

- [Windows] Use uplink MAC as source MAC when transmitting packets to underlay network from Windows Nodes. Security-related configurations like "Forged transmits" in VMware vSphere don't need to be degraded. ([#3516](https://github.com/antrea-io/antrea/pull/3516), [@wenyingd])
- Use iptables-wrapper in Antrea container to support distros that runs iptables in "nft" mode. ([#3276](https://github.com/antrea-io/antrea/pull/3276), [@antoninbas])

> - Install cni conf after installing cni bins to support container runtime cri-o. ([#3154](https://github.com/antrea-io/antrea/pull/3154), [@tnqn])

Reviewer: CNI configuration files

CNI binaries

Reviewer: `cri-o`

- Upgrade packaged Whereabouts version to v0.5.1. ([#3511](https://github.com/antrea-io/antrea/pull/3511), [@antoninbas])
- Upgrade to go-ipfix v0.5.12. ([#3352](https://github.com/antrea-io/antrea/pull/3352), [@yanjunz97])
- Upgrade Kustomize from v3.8.8 to v4.4.1 to fix Cronjob patching bugs. ([#3402](https://github.com/antrea-io/antrea/pull/3402), [@yanjunz97])
- Fail in Agent initialization if GRE tunnel type is used with IPv6. ([#3156](https://github.com/antrea-io/antrea/pull/3156), [@antoninbas])
- Refactor the OpenFlow pipeline for future extensibility. ([#3058](https://github.com/antrea-io/antrea/pull/3058), [@hongliangl])
- Validate IP ranges of IPPool for Antrea IPAM. ([#2995](https://github.com/antrea-io/antrea/pull/2995), [@ksamoray])

> - Validate values of protocol and labelSelector in the crd schema of Antrea-native policies. ([#3342](https://github.com/antrea-io/antrea/pull/3342), [@KMAnju-2021])

Reviewer: CRD

- Validate labels in ANP/ACNP/CG. ([#3331](https://github.com/antrea-io/antrea/pull/3331), [@GraysonWu])
- Reduce permission of antrea service accounts. ([#3393](https://github.com/antrea-io/antrea/pull/3393), [@tnqn])
- Remove Kind-specific manifest and scripts. Antrea now uses OVS kernel datapath for Kind clusters. ([#3413](https://github.com/antrea-io/antrea/pull/3413), [@antoninbas])
- Remove --k8s-1.15 flag from hack/generate-manifest.sh. ([#3350](https://github.com/antrea-io/antrea/pull/3350), [@antoninbas])
- Remove all legacy (*.antrea.tanzu.vmware.com) APIs. ([#3299](https://github.com/antrea-io/antrea/pull/3299), [@antoninbas])
- Remove unnecessary CRDs and RBAC rules from Multi-cluster manifest. ([#3491](https://github.com/antrea-io/antrea/pull/3491), [@luolanzone])
- Update label and image repo of antrea-mc-controller to be consistent with antrea-controller and antrea-agent. ([#3266](https://github.com/antrea-io/antrea/pull/3266) [#3466](https://github.com/antrea-io/antrea/pull/3466), [@luolanzone])

> - Add clusterID annotation to ServiceExport/Import resources. ([#3359](https://github.com/antrea-io/antrea/pull/3359), [@luolanzone])

Reviewer: I feel this one is minor to include in the change log. No strong opinion.

Author: keep it as it's a change to API resources.

Reviewer: Maybe next time we should group items of a feature if more than 3 items for the feature. It is not easy to understand individual small changes.

- Do not log error when Service for Endpoints is not found to avoid log spam. ([#3256](https://github.com/antrea-io/antrea/pull/3256), [@tnqn])

> - Ignore Services of type ExternalName in NodePortLocal Controller. ([#3114](https://github.com/antrea-io/antrea/pull/3114), [@antoninbas])

Reviewer: "in NodePortLocal Controller" -> "for NodePortLocal feature"

> - Add powershell command replacement in windows doc. ([#3264](https://github.com/antrea-io/antrea/pull/3264), [@GraysonWu])

Reviewer: the Antrea Windows documentation

### Fixed

> - Add userspace ARP/NDP responders to fix Egress and ServiceExternalIP supports for IPv6 clusters. ([#3318](https://github.com/antrea-io/antrea/pull/3318), [@hty690])

Reviewer: supports -> support

> - Add an agent config parameter "enableBridgingMode" for enabling flexible IPAM (bridging mode). ([#3297](https://github.com/antrea-io/antrea/pull/3297) [#3365](https://github.com/antrea-io/antrea/pull/3365), [@jianjuns])

Reviewer: I feel this is a change.

Author: right, added it here by mistake

- Fix incorrect results by `antctl get networkpolicy` when both Pod and Namespace are specified. ([#3499](https://github.com/antrea-io/antrea/pull/3499), [@Dyanngg])
- Fix IP leak issue when AntreaIPAM is enabled. ([#3314](https://github.com/antrea-io/antrea/pull/3314), [@gran-vmv])
- Fix error when dumping OVS flows of a NetworkPolicy via `antctl get ovsflows`. ([#3335](https://github.com/antrea-io/antrea/pull/3335), [@jainpulkit22])

> - Fix IPsec for IPv6 overlays. ([#3155](https://github.com/antrea-io/antrea/pull/3155), [@antoninbas])

Reviewer: IPsec encryption

- Add ignored interfaces names when getting interface by IP to fix NetworkPolicyOnly mode in AKE. ([#3219](https://github.com/antrea-io/antrea/pull/3219), [@wenyingd])
- Fix duplicate IP case for NetworkPolicy. ([#3467](https://github.com/antrea-io/antrea/pull/3467), [@tnqn])
- Don't delete the routes which are added for the peer IPv6 gateways on Agent startup. ([#3336](https://github.com/antrea-io/antrea/pull/3336) [#3490](https://github.com/antrea-io/antrea/pull/3490), [@Jexf] [@xliuxu])
- Fix pkt mark conflict between HostLocalSourceMark and SNATIPMark. ([#3430](https://github.com/antrea-io/antrea/pull/3430), [@tnqn])
- Unconditionally sync CA cert for Controller webhooks to fix Egress support when AntreaPolicy is disabled. ([#3421](https://github.com/antrea-io/antrea/pull/3421), [@antoninbas])

> - Fix inability to access NodePort in particular case. ([#3371](https://github.com/antrea-io/antrea/pull/3371), [@hongliangl])

Reviewer: case -> cases

> - Fix ipBlock referenced in nested ClusterGroup not processed correctly. ([#3383](https://github.com/antrea-io/antrea/pull/3383), [@Dyanngg])

Reviewer: ipBlocks

were not processed

- Realize Egress for a Pod once its network is created. ([#3360](https://github.com/antrea-io/antrea/pull/3360), [@tnqn])
- Fix NodePort/LoadBalancer issue when proxyAll is enabled. ([#3295](https://github.com/antrea-io/antrea/pull/3295), [@hongliangl])
- Do not panic when processing a PacketIn message for a denied connection. ([#3447](https://github.com/antrea-io/antrea/pull/3447), [@antoninbas])
- Fix CT mark matching without range in flow exporter. ([#3348](https://github.com/antrea-io/antrea/pull/3348), [@hongliangl])

> - [Windows] Enable IP forwarding of the Windows bridge local interface to fix support for Service of LoadBalancer type. ([#3137](https://github.com/antrea-io/antrea/pull/3137), [@hongliangl])

Reviewer: Service of tye LoadBalancer

# Changelog 1.6

## 1.6.0 - 2022-03-29

The Egress feature is graduated from Alpha to Beta and is therefore enabled by default.

### Added

- Add support for the following capabilities for [Antrea IPAM] feature:
  * Support pre-allocating continuous IPs for StatefulSet. ([#3281](https://github.com/antrea-io/antrea/pull/3281), [@annakhm])
  * Support specifying VLAN for IPPool. Traffic from Pods whose IPPools are configured with a VLAN ID will be tagged when leaving the Node uplink. ([#3247](https://github.com/antrea-io/antrea/pull/3247), [@gran-vmv])

> - Add support for the following capabilities for [Antrea Multi-cluster] feature:

Reviewer 4: nit:
```suggestion
- Add the following capabilities to the [Antrea Multi-cluster] feature:
```

  * Support Antrea ClusterNetworkPolicy replication in a ClusterSet. ([#3363](https://github.com/antrea-io/antrea/pull/3363), [@Dyanngg])
  * Add `antctl mc` subcommand for Antrea Multi-cluster resources. ([#3341](https://github.com/antrea-io/antrea/pull/3341) [#3287](https://github.com/antrea-io/antrea/pull/3287), [@hjiajing] [@bangqipropel])

> - Add support for the following capabilities for [AntreaPolicy] feature:

Reviewer 4: same as above
```suggestion
- Add the following capabilities to the [AntreaPolicy] feature:
```

>   * Add Node selector in Antrea-native policies to allow matching traffic originating from or destined for Nodes. ([#3038](https://github.com/antrea-io/antrea/pull/3038), [@wenqiq])

Reviewer 4: s/originating from or destined for Nodes/originating from specific Nodes or destined to specific Nodes

  * Add ServiceAccount selector in Antrea-native policies to allow selecting Pods by ServiceAccount. ([#3044](https://github.com/antrea-io/antrea/pull/3044), [@GraysonWu])

>   * Support Pagination for ClusterGroupMembership. ([#3183](https://github.com/antrea-io/antrea/pull/3183), [@qiyueyao])

Reviewer 4: maybe: Support Pagination for ClusterGroupMembership API

  * Add Port Number to Audit Logging. ([#3277](https://github.com/antrea-io/antrea/pull/3277), [@qiyueyao])
- [Flow Visibility] Add Grafana Flow Collector as the new visualization tool for flow records.
  * Add Grafana dashboards, Clickhouse data schema, deployment files, and doc. ([#3063](https://github.com/antrea-io/antrea/pull/3063) [#3525](https://github.com/antrea-io/antrea/pull/3525), [@heanlan] [@zyiou] [@dreamtalen])
  * Add support for exporting flow records to ClickHouse from Flow Aggregator. ([#3196](https://github.com/antrea-io/antrea/pull/3196) [#3526](https://github.com/antrea-io/antrea/pull/3526), [@wsquan171] [@dreamtalen])
  * Add ClickHouse monitor to ensure data retention for in-memory ClickHouse deployment. ([#3244](https://github.com/antrea-io/antrea/pull/3244) [#3498](https://github.com/antrea-io/antrea/pull/3498), [@yanjunz97])
- [Multicast] Support IGMPv3 leave action. ([#3389](https://github.com/antrea-io/antrea/pull/3389), [@wenyingd])
- [Windows] Add support for EndpointSlices on Windows Nodes. ([#3321](https://github.com/antrea-io/antrea/pull/3321), [@XinShuYang])

> - Add SKIP_CNI_BINARIES environment variable to support skipping installing specified CNI plugins. ([#3454](https://github.com/antrea-io/antrea/pull/3454), [@jainpulkit22])

Reviewer 4: skipping the installation of specified CNI plugins

> - Support antrea-agent UBI8 based image. ([#3273](https://github.com/antrea-io/antrea/pull/3273), [@ksamoray])

Reviewer 4: technically we have a single image, so it's not specific to the antrea-agent, it can be used to run the controller as well

so maybe: Support UBI8-based container image to run Antrea

> - Support UBI8-based container image to run Antrea. ([#3273](https://github.com/antrea-io/antrea/pull/3273), [@ksamoray])

Reviewer 3: nit: The above can be tagged with [Openshift] Since UBI8 images are specifically used for that distro (in theory you can use them anywhere, but in Openshift you can only use them)

- Add the following documentations:
  * Add [documentation](https://github.com/antrea-io/antrea/blob/v1.6.0/docs/service-loadbalancer.md) for ServiceExternalIP feature and Service of type LoadBalancer. ([#3322](https://github.com/antrea-io/antrea/pull/3322), [@hty690])
  * Add [documentation](https://github.com/antrea-io/antrea/blob/v1.6.0/docs/minikube.md) for deploying Antrea to Minikube cluster. ([#3391](https://github.com/antrea-io/antrea/pull/3391), [@jainpulkit22])
  * Add [documentation](https://github.com/antrea-io/antrea/blob/v1.6.0/docs/multicluster/antctl.md) for `antctl` Multi-cluster commands. ([#3414](https://github.com/antrea-io/antrea/pull/3414), [@bangqipropel])
  * Add [documentation](https://github.com/antrea-io/antrea/blob/v1.6.0/docs/antrea-ipam.md) for Multiple-VLAN support. ([#3507](https://github.com/antrea-io/antrea/pull/3507), [@gran-vmv])
  * Add [upgrade guide](https://github.com/antrea-io/antrea/blob/v1.6.0/docs/multicluster/upgrade.md) for Multi-cluster. ([#3374](https://github.com/antrea-io/antrea/pull/3374), [@luolanzone])

>   * Add [a per-rule example](https://github.com/antrea-io/antrea/blob/v1.6.0/docs/feature-gates.md#networkpolicystats) for NetworkpolicyStats docs . ([#3356](https://github.com/antrea-io/antrea/pull/3356), [@ceclinux])

Reviewer 4: there is an extra space before the period at the end of the line

### Changed

> - [Windows] Use uplink MAC as source MAC when transmitting packets to underlay network from Windows Nodes. Security-related configurations like "Forged transmits" in VMware vSphere don't need to be degraded. ([#3516](https://github.com/antrea-io/antrea/pull/3516), [@wenyingd])

Reviewer 4: I am not sure what you mean by "degraded" here?

Author: I wanted to say securty level degraded but guess it was unclear, how about the following?
> MAC address spoofing configurations like "Forged transmits" in VMware vSphere don't need to be enabled"

- Fail in Agent initialization if GRE tunnel type is used with IPv6. ([#3156](https://github.com/antrea-io/antrea/pull/3156), [@antoninbas])
- Refactor the OpenFlow pipeline for future extensibility. ([#3058](https://github.com/antrea-io/antrea/pull/3058), [@hongliangl])
- Validate IP ranges of IPPool for Antrea IPAM. ([#2995](https://github.com/antrea-io/antrea/pull/2995), [@ksamoray])
- Validate protocol in the CRD schema of Antrea-native policies. ([#3342](https://github.com/antrea-io/antrea/pull/3342), [@KMAnju-2021])
- Validate labels in the CRD schema of Antrea-native policies and ClusterGroup. ([#3331](https://github.com/antrea-io/antrea/pull/3331), [@GraysonWu])
- Reduce permissions of Antrea ServiceAccounts. ([#3393](https://github.com/antrea-io/antrea/pull/3393), [@tnqn])
- Remove Kind-specific manifest and scripts. Antrea now uses OVS kernel datapath for Kind clusters. ([#3413](https://github.com/antrea-io/antrea/pull/3413), [@antoninbas])
- Remove --k8s-1.15 flag from hack/generate-manifest.sh. ([#3350](https://github.com/antrea-io/antrea/pull/3350), [@antoninbas])

> - Remove all legacy (*.antrea.tanzu.vmware.com) APIs. ([#3299](https://github.com/antrea-io/antrea/pull/3299), [@antoninbas])

Reviewer 4: I don't know if you ordered these changes based on user impact, but it feels like this item should be higher up the list

Author: Agree, moved 3299 and 3413 to top of the section

- Add clusterID annotation to ServiceExport/Import resources. ([#3359](https://github.com/antrea-io/antrea/pull/3359), [@luolanzone])
- Do not log error when Service for Endpoints is not found to avoid log spam. ([#3256](https://github.com/antrea-io/antrea/pull/3256), [@tnqn])
- Ignore Services of type ExternalName for NodePortLocal feature. ([#3114](https://github.com/antrea-io/antrea/pull/3114), [@antoninbas])
- Add powershell command replacement in the Antrea Windows documentation. ([#3264](https://github.com/antrea-io/antrea/pull/3264), [@GraysonWu])

### Fixed

- Add userspace ARP/NDP responders to fix Egress and ServiceExternalIP support for IPv6 clusters. ([#3318](https://github.com/antrea-io/antrea/pull/3318), [@hty690])
- Fix incorrect results by `antctl get networkpolicy` when both Pod and Namespace are specified. ([#3499](https://github.com/antrea-io/antrea/pull/3499), [@Dyanngg])
- Fix IP leak issue when AntreaIPAM is enabled. ([#3314](https://github.com/antrea-io/antrea/pull/3314), [@gran-vmv])

> - Fix error when dumping OVS flows of a NetworkPolicy via `antctl get ovsflows`. ([#3335](https://github.com/antrea-io/antrea/pull/3335), [@jainpulkit22])

Reviewer 3: pedant nit: consider replacing "of a" with "for a"

Author: done

- Fix IPsec encryption for IPv6 overlays. ([#3155](https://github.com/antrea-io/antrea/pull/3155), [@antoninbas])
- Add ignored interfaces names when getting interface by IP to fix NetworkPolicyOnly mode in AKE. ([#3219](https://github.com/antrea-io/antrea/pull/3219), [@wenyingd])
- Fix duplicate IP case for NetworkPolicy. ([#3467](https://github.com/antrea-io/antrea/pull/3467), [@tnqn])
- Don't delete the routes which are added for the peer IPv6 gateways on Agent startup. ([#3336](https://github.com/antrea-io/antrea/pull/3336) [#3490](https://github.com/antrea-io/antrea/pull/3490), [@Jexf] [@xliuxu])
- Fix pkt mark conflict between HostLocalSourceMark and SNATIPMark. ([#3430](https://github.com/antrea-io/antrea/pull/3430), [@tnqn])
- Unconditionally sync CA cert for Controller webhooks to fix Egress support when AntreaPolicy is disabled. ([#3421](https://github.com/antrea-io/antrea/pull/3421), [@antoninbas])
- Fix inability to access NodePort in particular cases. ([#3371](https://github.com/antrea-io/antrea/pull/3371), [@hongliangl])

> - Fix ipBlocks referenced in nested ClusterGroup were not processed correctly. ([#3383](https://github.com/antrea-io/antrea/pull/3383), [@Dyanngg])

Reviewer 3: pedant nit: "were" in the sentence above seems redundant

Author: I also thought this is not needed, added it according Jianjun's [comment](https://github.com/antrea-io/antrea/pull/3531#discussion_r836702152). Removed it now. @jianjuns please let us know if you believe it's needed.

> - Realize Egress for a Pod once its network is created. ([#3360](https://github.com/antrea-io/antrea/pull/3360), [@tnqn])

Reviewer 4: Realize Egress for a Pod as soon as its network is created.
